Core/src/Definitions/ParticleData.cpp

```diff
@@ -23,77 +25,129 @@
 
 namespace {
 
-// TODO the following functions could be constexpr but we are currently limited
-// by `std::find`
-
-static inline std::optional<std::size_t> findIndexByPdg(std::int32_t pdg) {
-  auto beg = std::cbegin(kParticlesPdgNumber);
-  auto end = std::cend(kParticlesPdgNumber);
-  // assumes sorted container of pdg numbers
-  auto pos = std::find(beg, end, pdg);
-  if (pos == end) {
-    return std::nullopt;
-  }
-  return std::make_optional(std::distance(beg, pos));
+enum class Type {
+  Charge,
+  Mass,
+  Name,
+};
+
+// Template function depending on the PDG and *type* of data
+// so we can use statically cached values
+template <typename T, Acts::PdgParticle pdg, Type type>
+std::optional<T> findCachedImpl(const std::map<std::int32_t, T>& map) {
+  const static std::optional<T> value = [&map]() -> std::optional<T> {
+    const auto it = map.find(pdg);
+    if (it == map.end()) {
+      return std::nullopt;
+    }
+    return it->second;
+  }();
+
+  return value;
```

Comment:
is the lambda necessary for type deduction? other I cannot really see a benefit just looking at the code

Reply:
It's used as a way to initialize a `static const` with more than a simple expression.

Reply:
ah of course - sorry missed the `static`